src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -52,342 +56,3008 @@ const ActorInit Boss_Ganon2_InitVars = {
+            this->unk_3A4.x = (player->actor.world.pos.x - 40.0f) - 200.0f;
+            this->unk_3A4.y = (player->actor.world.pos.y + 40.0f) - 30.0f;
+            this->unk_3A4.z = (player->actor.world.pos.z - 20.0f) + 100.0f;
+            this->unk_3B0.x = player->actor.world.pos.x;
+            this->unk_3B0.y = ((player->actor.world.pos.y + 10.0f + 60.0f) - 20.0f) + 30.0f;
+            this->unk_3B0.z = player->actor.world.pos.z;
+            this->unk_3BC.x = 0.8f;
+            if (this->actor.world.pos.y <= 1099.0f) {
+                this->actor.world.pos.y = 1099.0f;
+                this->unk_39C = 0x15;
+                this->unk_398 = 0;
+                this->unk_420 = 10.0f;
+                this->actor.velocity.y = 0.0f;
+                Animation_MorphToPlayOnce(&this->skelAnime, &object_ganon_anime3_Anim_010380, 0.0f);
+                func_808FD4D4(this, globalCtx, 0, 3);
+                func_800A9F6C(0.0f, 0xC8, 0x14, 0x14);
+            }
+            break;
+        case 21:
+            this->unk_339 = 6;
+            SkelAnime_Update(&this->skelAnime);
+            this->unk_41C = Math_CosS(globalCtx->gameplayFrames << 0xF) * this->unk_420;
+            Math_ApproachZeroF(&this->unk_420, 1.0f, 0.75f);
+            if (this->unk_398 == 0x1E) {
+                this->unk_39C = 0x16;
+                this->unk_30C = 10.0f;
+            } else {
+                break;
+            }
+        case 22:
+            if (this->unk_398 < 0x3C) {
+                this->unk_339 = 7;
+            }
+            this->unk_3BC.x = 0.0f;
+            this->actor.world.pos.y = 1099.0f;
+            SkelAnime_Update(&this->skelAnime);
+            Math_ApproachZeroF(&this->unk_30C, 1.0f, 0.1f);
+            if (this->unk_398 >= 0x33) {
+                Math_ApproachF(&this->unk_224, 1.0f, 1.0f, 0.025f);
+            }
+            if (this->unk_398 == 0x3C) {
+                this->unk_336 = 2;
+            }
+            if (this->unk_398 == 0x50) {
+                BossGanon2_SetObjectSegment(this, globalCtx, OBJECT_GANON2, false);
+                TitleCard_InitBossName(globalCtx, &globalCtx->actorCtx.titleCtx, SEGMENTED_TO_VIRTUAL(D_06021A90), 160,
+                                       180, 128, 40);
+            }
+            this->unk_3A4.x = ((this->actor.world.pos.x + 500.0f) - 350.0f) + 100.0f;
+            this->unk_3A4.y = this->actor.world.pos.y;
+            this->unk_3A4.z = this->actor.world.pos.z;
+            this->unk_3B0.x = this->actor.world.pos.x;
+            this->unk_3B0.z = this->actor.world.pos.z;
+            this->unk_3B0.y = (this->unk_1B8.y + 60.0f) - 40.0f;
+            if (this->unk_398 > 0xA6 && this->unk_398 < 0xAD) {
+                this->unk_312 = 2;
+            }
+            if (this->unk_398 > 0xBA && this->unk_398 < 0xC4) {
+                this->unk_312 = 1;
+            }
+            if (this->unk_398 > 0xCA && this->unk_398 < 0xD2) {
+                this->unk_312 = 2;
+            }
```

Comment:
i think 398 is a timer value and all these can be decimal. but if youd rather wait for a documentation pass I think that probably fine too

Reply:
though itd probably make the cutscene functions easier to read in the mean time while things arent documented. so i do recommend it.

you can see a bunch of timers being decremented in the update function. Id try to make all uses of those variables decimal

Reply:
this is the last thing on my end and itd be good to go from me. Having them in decimal already will make docs alot easier down the road I think, so we should do it now.

I know of a good vscode extension that lets you highlight numbers and convert them between dec and hex, if youd like

Reply:
Haha I uh just finished doing it all manually and it was pretty awful but I will keep that in mind for next time. Let me know if I missed anything.

Reply:
nice lol
the plugin is NumberMonger. you highlight the number then press ctrl+p to bring up the command palette, and type "convert" and youll see a bunch of conversion options. Idk if its possible to bind it to a shortcut to skip the palette, prob is.

Its still a bit manual, but you at least dont need to tab out and go to a calculator everytime